pkg/agent/util/net.go

```diff
@@ -19,28 +19,37 @@ import (
 	"encoding/hex"
 	"fmt"
 	"io"
-	"strings"
 )
 
 const (
 	interfaceNameLength   = 15
-	podNamePrefixLength   = 8
-	containerKeyConnector = `-`
+	interfacePrefixLength = 8
+	interfaceKeyLen       = interfaceNameLength - (interfacePrefixLength + 1)
 )
 
-// Calculates a suitable interface name using the pod namespace and pod name. The output should be
-// deterministic (so that multiple calls to GenerateContainerInterfaceName with the same parameters
-// return the same value). The output should have length interfaceNameLength (15). The probability of
-// collision should be neglectable.
-func GenerateContainerInterfaceName(podName string, podNamespace string) string {
+func generateInterfaceName(id string, prefix string) string {
 	hash := sha1.New()
-	podID := fmt.Sprintf("%s/%s", podNamespace, podName)
-	io.WriteString(hash, podID)
-	podKey := hex.EncodeToString(hash.Sum(nil))
-	name := strings.Replace(podName, "-", "", -1)
-	if len(name) > podNamePrefixLength {
-		name = name[:podNamePrefixLength]
+	io.WriteString(hash, id)
+	interfaceKey := hex.EncodeToString(hash.Sum(nil))
+	if len(prefix) > interfacePrefixLength {
+		prefix = prefix[:interfacePrefixLength]
 	}
-	podKeyLength := interfaceNameLength - len(name) - len(containerKeyConnector)
-	return strings.Join([]string{name, podKey[:podKeyLength]}, containerKeyConnector)
+	return fmt.Sprintf("%s-%s", prefix, interfaceKey[:interfaceKeyLen])
+}
+
+// GenerateContainerInterfaceName generates a unique interface name using the
+// Pod's Namespace and name. The output should be deterministic (so that
+// multiple calls to GenerateContainerInterfaceName with the same parameters
+// return the same value). The output has the length of interfaceNameLength(15).
+// The probability of collision should be neglectable.
+func GenerateContainerInterfaceName(podName string, podNamespace string) string {
```

Comment:
Would it cause existing interface names change? Would upgrading be affected?

Reply:
A good question. I think it might be ok not to support upgrade yet. What you think?

Reply:
Maybe I should first ask which format you like.

To handle upgrade, we could change to use Pod name + Namespace name to be the key.

Reply:
Would not supporting upgrade make users hard to use new features and get bugs in their clusters fixed? I think handling upgrade in code to consider previously used key might be complicated and hard to describe the compatibility in doc when we want to remove it. Do you think we can just use major release number to indicate compatibility like some software do? If it's 0.2.0 to 0.3.0, maybe we should avoid changes that break compatibility and leave it to major release change if it's necessary.

Reply:
I think it depends on how many deployments will use 0.2 and 0.1 and would expect upgrade without impact to existing Pods. If we want to make sure nothing breaks across versions, guess we need to add upgrade tests.

But for this change there is another way to handle it - if we believe not safe to use the generated interface name as the key we could instead use Pod + NS or Node name to be the key for InterfaceStore.

Reply:
Agree it's ok to ignore existing Pods at this stage. 
If it's just changing key of InterfaceStore, I think it doesn't break compatibility? My concern was GenerateContainerInterfaceName is used in setupInterfaces for host veth name, not sure what would happen if it changes. Anyway, I agree it's fine to not handle upgrade case until we declare it and test it.

Reply:
Thought over. Maybe let me revert the name change in this PR, and we could make a separate one to change it.
Do you support the new format I used?

Reply:
Is the difference keeping "-" in the name and different hashing value? For the former, I'm not sure why @wenyingd removed it and whether it's related to compatibility when names contain "-". If not, I'm fine with both.
@wenyingd could you confirm?

Reply:
I restored the original format.